lib/beaker/dsl/roles.rb

```diff
@@ -115,7 +115,11 @@ def not_controller(host)
       #       puts "this host is ONLY an agent!"
       #     end
       def agent_only(host)
-        host['roles'].length == 1 && host['roles'].include?('agent')
+        if host[:pe_ver] && version_is_less(host[:pe_ver], '4.0')
+          false
```

Comment:
the test "agent_only" becomes a bit ambiguous here, because there's a small chance someone might consider a "legacy agent" as an agent. This is admittedly a corner case!